src/KernelStack/HttpMiddleware.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace Pyrite\KernelStack;
+
+use Pyrite\Kernel\PyriteKernel;
+use Pyrite\Routing\RouteConfigurationBuilderI18n;
+use Pyrite\Routing\RouteConfigurationBuilderImpl;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\MethodNotAllowedHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\HttpKernel\TerminableInterface;
+use Pyrite\Routing\Director;
+use Symfony\Component\Routing\Exception\MethodNotAllowedException;
+use Symfony\Component\Routing\Exception\ResourceNotFoundException;
+use Symfony\Component\Routing\Matcher\UrlMatcher;
+use Symfony\Component\Routing\Matcher\UrlMatcherInterface;
+
+class HttpMiddleware implements HttpKernelInterface, TerminableInterface
+{
+    /**
+     * @var string
+     */
+    protected $routingConfigPath;
+
+    /**
+     * @var PyriteKernel
+     */
+    protected $kernel;
+
+    /**
+     * @var HttpKernelInterface
+     */
+    protected $app;
+
+    /**
+     * HttpMiddleware constructor.
+     *
+     * @param HttpKernelInterface $app
+     * @param PyriteKernel        $kernel
+     * @param                     $routingConfigPath
+     */
+    public function __construct(
+        HttpKernelInterface $app,
+        PyriteKernel $kernel,
+        $routingConfigPath
+    ) {
+        $this->app = $app;
+        $this->kernel = $kernel;
+        $this->routingConfigPath = $routingConfigPath;
+    }
+
+    /**
+     * @param Request $request
+     * @param int     $type
+     * @param bool    $catch
+     */
+    public function handle(Request $request, $type = self::MASTER_REQUEST, $catch = true)
+    {
+        // Create container and freeze config
+        $container = $this->kernel->startContainer();
+
+        $director = new Director($request, $this->routingConfigPath);
+
+        $config = $this->kernel->getConfig();
+
+        $currentLocal = $config->get('current_locale');
+        $availableLocal = $config->get('available_locales');
+
+        if(null !== $currentLocal && null !== $availableLocal){
```

Comment:
Choosing the RouteConfigurationBuilder shouldn't be handled by HttpMiddleware

currentLocal(e) & availableLocal(e) are Evaneos-centric, you shouldn't rely on them in the HttpMiddleware (supposed to be generic)